src/classrooms/classrooms.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable, NotFoundException, ForbiddenException } from "@nestjs/common";
+
+import { EntityManager } from "@mikro-orm/core";
+import { InjectRepository } from "@mikro-orm/nestjs";
+
+import { Classroom } from "@/common/entities/classrooms.entity";
+import { User } from "@/common/entities/users.entity";
+import { TeachersRepository } from "@/teachers/teachers.repository";
+
+import { CreateClassroomDto, ClassroomResponseDto } from "./classrooms.dtos";
+import { ClassroomsRepository } from "./classrooms.repository";
+
+@Injectable()
+export class ClassroomsService {
+  constructor(
+    @InjectRepository(Classroom)
+    private readonly classroomsRepository: ClassroomsRepository,
+    private readonly teacherRepository: TeachersRepository,
+    private readonly em: EntityManager,
+  ) {}
+
+  async createClassroom(
+    createClassroomDto: CreateClassroomDto,
+    authenticatedUserId: number,
+  ): Promise<ClassroomResponseDto> {
+    const { title, subject, classTime, days } = createClassroomDto;
+
+    const user = await this.em.findOne(
+      User,
+      { id: authenticatedUserId },
+      { populate: ["teacher"] },
+    );
+    if (!user) {
+      throw new NotFoundException("User not found");
+    }
+
+    if (!user.teacher) {
+      throw new ForbiddenException("Only teachers can create classrooms");
+    }
+
+    const classroom = new Classroom();
+    classroom.title = title;
+    classroom.subject = subject;
+    classroom.classTime = new Date(classTime);
+    classroom.days = days;
+    classroom.teacher = user.teacher;
```

Comment:
Let's not create new records like this. Should simply call the classroom repository createOne and pass the classroom dto.